low-level/sensors/impl/generic/node_sas_port.py

```diff
@@ -309,8 +405,25 @@ def _create_json_message(self, alert_type):
         specific_info = {}
         specific_info_list = []
 
+        if port != -1:
+            # This is a port level alert, add an error key in specific info
+            if alert_type == 'fault':
+                specific_info["error"] = "No connectivity detected on the port, possible \
```

Comment:
Add SAS port number here and update the event string as follows
               
> specific_info["error"] = "No connectivity detected on the SAS port %s, possible \
                                          causes could be missing SAS cable, bad cable connection, faulty cable or SAS port failure"